infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -102,45 +95,41 @@
 
     private ExecutorService executor;
 
-    @Override
-    public ExecutorService getExecutor(CorfuMsgType corfuMsgType) {
-        return executor;
-    }
-
-    @Override
-    public List<ExecutorService> getExecutors() {
-        return Collections.singletonList(executor);
-    }
-
     /**
      * Returns a new LogUnitServer.
      *
      * @param serverContext context object providing settings and objects
      */
     public LogUnitServer(ServerContext serverContext) {
         this.serverContext = serverContext;
-        this.config = LogUnitServerConfig.parse(serverContext.getServerConfig());
-        executor = Executors.newFixedThreadPool(serverContext.getLogunitThreadCount(),
+        executor = Executors.newFixedThreadPool(serverContext.getConfiguration().getNumLogUnitWorkerThreads(),
                 new ServerThreadFactory("LogUnit-", new ServerThreadFactory.ExceptionHandler()));
 
-        if (config.isMemoryMode()) {
+        if (serverContext.getConfiguration().getInMemoryMode()) {
             log.warn("Log unit opened in-memory mode (Maximum size={}). "
                     + "This should be run for testing purposes only. "
                     + "If you exceed the maximum size of the unit, old entries will be "
                     + "AUTOMATICALLY trimmed. "
                     + "The unit WILL LOSE ALL DATA if it exits.", Utils
-                    .convertToByteStringRepresentation(config.getMaxCacheSize()));
+                    .convertToByteStringRepresentation(serverContext.getConfiguration().getMaxLogUnitCacheSize()));
             streamLog = new InMemoryStreamLog();
         } else {
-            streamLog = new StreamLogFiles(serverContext, config.isNoVerify());
+            streamLog = new StreamLogFiles(serverContext);
         }
 
-        dataCache = new LogUnitServerCache(config, streamLog);
-        batchWriter = new BatchProcessor(streamLog, serverContext.getServerEpoch(), !config.isNoSync());
+        dataCache = new LogUnitServerCache(streamLog, serverContext.getConfiguration().getMaxLogUnitCacheSize());
+        batchWriter = new BatchProcessor(streamLog, serverContext.getServerEpoch(),
+                serverContext.getConfiguration().getSyncData());
 
         logCleaner = new StreamLogCompaction(streamLog, 10, 45, TimeUnit.MINUTES, ServerContext.SHUTDOWN_TIMER);
     }
 
+
+    @Override
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/app/CorfuDB/CorfuDB/pullRequest?prid=4662120)